### DISCUSSION TITLE
MET compile issue

### DISCUSSION THREAD

**Discussion by scatt11** [dtcenter/METplus#1157] [Installation] [answered]
Hello,
I am having trouble compiling MET 10.0.0.  I am using the compile_Met_all.sh script and it seems to be getting tripped up with the make of BUFRLIB.  Operating system is Ubuntu 18.04.5.  Attached you will find my input file, screen output messages, configure.log, config.log, and make.log.

[config.log](https://github.com/dtcenter/METplus/files/7165514/config.log)
[configure.log](https://github.com/dtcenter/METplus/files/7165515/configure.log)
[make.log](https://github.com/dtcenter/METplus/files/7165516/make.log)
[screen_output.txt](https://github.com/dtcenter/METplus/files/7165517/screen_output.txt)

Any assistance would be much appreciated.

**Comment by georgemccabe** [comment]
Hi @scatt11, I see you are having issues installing MET. I have assigned this issue to @jprestop, who is familiar with common errors in the installation process. Was a make.log file created in the bufrlib directory? If so, could you also attach this file?

/home/computeadmin/meteo/Libraries/MET_install/external_libs/bufrlib/BUFRLIB_v11.3.0/make.log

**Reply by scatt11** [reply]
Thanks for the quick reply @georgemccabe.  Yes a make.log was created for bufrlib and it is attached here.  Please let me know what other information you may need.

[bufrlib_make.log](https://github.com/dtcenter/METplus/files/7171831/bufrlib_make.log)

**Comment by jprestop** [comment, marked as answer]
Hi @scatt11.  I see that you are having trouble with the compilation of BUFRLIB in the compile_Met_all.sh script.  Thank you very much for all of the helpful log information you sent. 

I see the following in the make.log file for BUFRLIB:

`gfortran: error: unrecognized command line option ‘-fallow-argument-mismatch’; did you mean ‘-Wno-argument-mismatch’?`

‘-Wno-argument-mismatch' was deprecated in favor of ‘-fallow-argument-mismatch'.  It looks like that was deprecated with version 10 of the GNU compilers (https://gcc.gnu.org/gcc-10/changes.html).   I see in your config.log file that you are using 7.5.0 of the GNU compilers.  

Let's have you try changing the following line in the compile_MET_all.sh script from:

${FC} -c -fno-second-underscore **-fallow-argument-mismatch** \`./getdefflags_F.sh\` modv*.F moda*.F \`ls -1 *.F *.f | grep -v "mod[av]_"\` >> make.log 2>&1

to 

${FC} -c -fno-second-underscore **-Wno-argument-mismatch** \`./getdefflags_F.sh\` modv*.F moda*.F \`ls -1 *.F *.f | grep -v "mod[av]_"\` >> make.log 2>&1
 
Then, please try recompiling everything and seeing if that does the trick for BUFRLIB and for MET.  Please let us know how it goes.

I have a [GitHub issue](https://github.com/dtcenter/MET/issues/1395) set up to make improvement to the script, and I have just added this as a change to improve the script.  (If GNU version 10 or greater, have ‘-fallow-argument-mismatch' in BUFRLIB compilation. If less than 10, have ‘-Wno-argument-mismatch'' in BUFRLIB compilation.)

**Reply by scatt11** [reply]
@jprestop that did the trick! Thank you very much for the prompt support.
